config/config_test.go

```diff
@@ -62,17 +78,47 @@ func TestValidConfigs(t *testing.T) {
 func TestInvalidConfigs(t *testing.T) {
 	for k, v := range invalidConfigs {
 		t.Run(k, func(t *testing.T) {
-			filename := mkConfig(t, v)
-			_, err := ReadConfigFile(filename, "", "")
+			filename := mkConfig(t, t.TempDir(), v)
+			_, err := ReadConfigFile(filename, "", "", "")
 			if err == nil {
 				t.Fatal("this config must generate an error")
 			}
 		})
 	}
 }
 
-func mkConfig(t *testing.T, body string) string {
-	f, err := ioutil.TempFile(t.TempDir(), "config_test_*")
+func TestConfigDir(t *testing.T) {
+
+	dir := t.TempDir()
+	f := mkConfig(t, dir, validSubConfig["sub config1"])
```

Comment:
You could probably make this a bit cleaner and generalized by just iterating over the map:

```suggestion
	for configName, configData := range validSubConfig {
		f := mkConfig(t, dir, configData)
		t.Logf("created config for: %s: %s", configName, f)
	} 
```

Reply:
+ removing calls to mkConfig/t.Log bellow